components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/DefaultNotificationHandler.java

```diff
@@ -146,6 +145,11 @@ protected Map<String, String> buildNotificationData(Event event) throws Identity
             if (NotificationChannels.SMS_CHANNEL.getChannelType().equals(notificationChannel) && userClaims
                     .containsKey(NotificationConstants.SMSNotification.DEFAULT_SMS_NOTIFICATION_LOCALE)) {
                 locale = userClaims.get(NotificationConstants.SMSNotification.DEFAULT_SMS_NOTIFICATION_LOCALE);
+            } else if (Boolean.TRUE.equals(eventProperties.get(NotificationConstants.IS_FEDERATED_USER)) &&
```

Comment:
why is part is not inside else block. I think, this if condition for notification channel. 
Also give the prority to user claim. Lets check the locale from userclaim is null and resolve from properties.

Reply:
Then why we need this else if block. seems like redundant